src/_pytest/fixtures.py

```diff
@@ -811,6 +813,8 @@ def __init__(
         self.has_location = baseid is not None
         self.func = func
         self.argname = argname
+        if callable(scope):
+            scope = scope(self._fixturemanager.config)
```

Comment:
I suggest we pass `config` by keyword:

```python
scope = scope(config=self._fixturemanager.config)
```

Also we can probably catch an error here and provide a more meaningful message:

```python
scope = eval_scope_callable(argname, scope, config)
```

With `eval_scope_callable` a function in the same module:

```python
def eval_scope_callable(fixturename, scope_callable, config):
    try:
        result = scope_callable(config=config)
    except Exception:
        raise TypeError(
            "Error evaluating {} while defining fixture '{}'.\n"
            "Expected a function with the signature (*, config)".format(scope_callable, fixturename)
        )
    if not isinstance(result, str):
        fail(
            "Expected {} to return a 'str' while defining fixture '{}', but it returned:\n"
            "{!r}".format(scope_callable, fixturename, result),
            pytrace=False,
        )   
    return result
```

Reply:
Also, should we pass other parameters while at it? `argname` (as `fixture_name`) and `item_id`? 🤔

Reply:
Thanks for the feedback. About the missing things in the PR, I am aware of them, but I just wanted to show my code first, to make sure that my approach is correct.  

I'll implement the `eval_scope_callable` as you suggested, but I'm not sure if passing the config as a keyword variable is necessary. For example, if the user wants to control scope via an environment variable, he doesn't need the config at all. The dynamic scope function could be defined as `def func(_): ...`, which will be better for pylint than unused argument.

Reply:
> Also, should we pass other parameters while at it? `argname` (as `fixture_name`) and `item_id`? 🤔

The `argname` makes sense to me, but not sure what you mean by `item_id`. Is it the `base_id` argument to `FixtureDef` init?

Reply:
Oh nevermind about `item_id`, we are talking about a fixture definition, not fixture invocation 🤦‍♂ 

Can we add `fixture_name` then? I feel it is a no-brainer.

Reply:
Sure, the `fixture_name` makes sense. I'll try to add it today.